dissect/target/plugins/os/windows/dpapi/dpapi.py

```diff
@@ -1,162 +1,167 @@
+
+        # Search for user master keys
+        #
+        # Generally located at $HOME/AppData/Roaming/Microsoft/Protect/{user_sid}/{mk_guid}
+        PROTECT_DIRS = [
+            # Windows Vista and newer
+            "AppData/Roaming/Microsoft/Protect",
+            # Windows XP
+            "Application Data/Microsoft/Protect",
+        ]
+
+        # TODO: We should probably identify users with their SID instead of username.
         for user in self.target.user_details.all_with_home():
-            path = user.home_path.joinpath("AppData/Roaming/Microsoft/Protect").joinpath(user.user.sid)
-            user_mks = self._load_master_keys_from_path(user.user.name, path)
-            if user_mks:
-                result[user.user.name] = user_mks
+            master_keys.setdefault(user.user.name, {})
 
-        return result
+            for protect_dir in PROTECT_DIRS:
+                path = user.home_path.joinpath(protect_dir).joinpath(user.user.sid)
+                if user_mks := self._load_master_keys_from_path(user.user.name, path):
+                    master_keys[user.user.name] |= user_mks
+
+        return master_keys
 
     @cached_property
     def _users(self) -> dict[str, dict[str, str]]:
         return {u.name: {"sid": u.sid} for u in self.target.users()}
 
-    def _load_master_keys_from_path(self, username: str, path: Path) -> dict[str, MasterKeyFile]:
+    def _load_master_keys_from_path(self, username: str, path: Path) -> Iterator[tuple[str, MasterKeyFile]]:
+        """Iterate over the provided ``path`` and search for master key files for the given user."""
+
         if not path.exists():
-            return {}
+            self.target.log.info(f"Unable to load master keys from path {path}: does not exist")
+            return
 
-        result = {}
         for file in path.iterdir():
-            if self.MASTER_KEY_REGEX.findall(file.name):
-                with file.open() as fh:
-                    mkf = MasterKeyFile(fh)
-
-                if username == self.SYSTEM_USERNAME:
-                    dpapi_system = CredSystem(self.secrets["DPAPI_SYSTEM"][16:])
-
-                    if not mkf.decrypt_with_key(dpapi_system.machine_key):
-                        mkf.decrypt_with_key(dpapi_system.user_key)
-
-                    # This should not be possible, decrypting the System master key should always succeed
-                    if not mkf.decrypted:
-                        raise Exception("Failed to decrypt System master key")
-
-                if user := self._users.get(username):
-                    for mk_pass in self.keychain():
+            if not self.MASTER_KEY_REGEX.findall(file.name):
+                continue
+
+            with file.open() as fh:
+                mkf = MasterKeyFile(fh)
+
+            # Decrypt SYSTEM master key using the DPAPI_SYSTEM LSA secret.
+            if username == self.SYSTEM_USERNAME:
+                if not self.target.lsa._secrets.get("DPAPI_SYSTEM"):
```

Comment:
```suggestion
                if "DPAPI_SYSTEM" not in self.target.lsa._secrets:
```
Additionally, `target.lsa._secrets` will raise an error if `ntversion` can't be retrieved, is that fine?